src/main/java/sungshin/sooon/controller/BoardController.java

```diff
@@ -0,0 +1,66 @@
+package sungshin.sooon.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.data.domain.Pageable;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import sungshin.sooon.dto.PostRequestDto;
+import sungshin.sooon.dto.PostResponseDto;
+import sungshin.sooon.dto.PostUpdateRequestDto;
+import sungshin.sooon.model.Post;
+import sungshin.sooon.service.BoardService;
+
+import java.security.Principal;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/board")
+public class BoardController {
+
+    private final BoardService boardService;
+
+    // 전체 글 리스트
+    @GetMapping("/list")
+    public Page<Post> list(@PageableDefault(page = 1, sort = "created_at", direction = Sort.Direction.DESC) Pageable pageable, Model model) {
+        Page<Post> postList = boardService.getPostList(pageable);
+        model.addAttribute("postList", postList);
+
+        return boardService.getPostList(pageable);
+    }
+
+    // 게시글 저장
+    @PostMapping("/post")
+    public Long createPost(@RequestBody PostRequestDto postRequestDto) {
+        return boardService.savePost(postRequestDto);
+    }
+
+    // 게시글 상세
+    @GetMapping("/detail/{post_id}")
```

Comment:
```suggestion
    @GetMapping("/{post_id}")
```

> 2.4. 행위(method)는 URL에 포함하지 않는다.
Bad
POST http://api.test.com/users/1/delete-post/1
Good
DELETE http://api.test.com/users/1/posts/1
출처: https://sanghaklee.tistory.com/57 [이상학의 개발블로그]